.github/workflows/integration-test-on-pr.yml

```
@@ -0,0 +1,120 @@
+name: integration-tests
+
+on:
+  pull_request:
+
+env:
+  NPL_SERVICE_ACCOUNT_CLIENT_SECRET: "${{ secrets.NPL_SERVICE_ACCOUNT_CLIENT_SECRET }}"
+
+jobs:
+  deploy-to-local:
+    name: Deploy to Local NOUMENA Runtime
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Get NPL CLI
+        shell: bash
+        run: |
+          set -euo pipefail
+          export NPL_CLI_INSTALL_DIR="$HOME/.local/bin"
+          mkdir -p "$NPL_CLI_INSTALL_DIR"
+          curl -s https://documentation.noumenadigital.com/get-npl-cli.sh | bash
+          echo "$NPL_CLI_INSTALL_DIR" >> "$GITHUB_PATH"
+          echo "$HOME/.npl/bin" >> "$GITHUB_PATH"
+          echo "$HOME/.noumena/bin" >> "$GITHUB_PATH"
+          export PATH="$NPL_CLI_INSTALL_DIR:$HOME/.npl/bin:$HOME/.noumena/bin:$PATH"
+          hash -r
+          command -v npl >/dev/null
+
+      - name: Start NOUMENA Runtime
+        shell: bash
+        run: |
+          set -euxo pipefail
+          docker compose build
+          docker compose up -d --wait
+
+      - name: Deploy NPL locally
+        shell: bash
+        env:
+          NOUMENA_SKIP_UPDATE_CHECK: "1"
+        run: |
+          set -euo pipefail
+          tmp_deploy="$(mktemp)"
+          npl deploy | tee "$tmp_deploy"
+          if ! grep -Fq "Successfully deployed NPL sources and migrations" "$tmp_deploy"; then
+            echo "::error::Expected success message not found in deployment output."
+            exit 1
+          fi
+
+      - name: Install frontend dependencies
+        working-directory: webapp
+        run: npm install
+
+      - name: Run Vite dev server smoke test
+        working-directory: webapp
+        shell: bash
+        run: |
+          set -euxo pipefail
+          npm run dev -- --host &
+          DEV_PID=$!
+          sleep 15
+          kill $DEV_PID || true
+          wait $DEV_PID || true
+
+      - name: Tear down NOUMENA Runtime
+        shell: bash
+        run: |
+          set -euxo pipefail
+          docker compose down --volumes --remove-orphans
```

I think that this step will not run if any of the previous steps in the job fails. Hece, the tear down would not happen if e.g. `Run Vite dev server smoke test` or `Deploy NPL locally` fails. What I think could be done is the folloing: 
```suggestion
      - name: Tear down NOUMENA Runtime
        if: always()
        shell: bash
        run: |
          set -euxo pipefail
          docker compose down --volumes --remove-orphans
```
`if: always()` ensures it runs regardless of whether the job succeeds or fails